Makefile

```diff
@@ -185,34 +194,23 @@ cadence-canary: $(ALL_SRC)
 
 go-generate-format: go-generate fmt
 
-go-generate:
-	GO111MODULE=off go get -u github.com/myitcv/gobin
-	GOOS= GOARCH= gobin -mod=readonly github.com/golang/mock/mockgen
-	GOOS= GOARCH= gobin -mod=readonly github.com/dmarkham/enumer
+go-generate: $(MOCKGEN_BIN) $(ENUMER_BIN)
```

Comment:
This is probably not related to this refactoring, but I have tried running `make go-generate` which regenerates mock files without licence headers and then runs `copyright` with `--verifyOnly` which then complains. Making this command kind of useless.

This is not true for generated thrift files though.  Looks like licengen handles `.gen` files differently: https://github.com/uber/cadence/blob/makethrift/cmd/tools/copyright/licensegen.go#L238

Reply:
yeah, this command will go away eventually I think.  it's fast enough to make thrift/proto re-generate _correctly_ when needed (which is not part of this PR), which makes an explicit command rather pointless.

as far as the `copyright` tool: what is the _desired_ behavior?  is there any reason to _not_ run it all the time without `--verifyOnly` so it prepends it anywhere it's missing?

Reply:
Yeah, I usually regenerated mock files as needed (within the IDE). Plus this runs very long anyway.

Reply:
ah, right, this is mockgen not protoc-gen.
yeah, that makes sense (running as-needed and where-needed).  if the mocks are out of date they'll fail tests anyway, so there's no real harm in leaving it manual.